ethers-providers/src/provider.rs

```diff
@@ -272,36 +272,43 @@ impl<P: JsonRpcClient> Middleware for Provider<P> {
             }
         }
 
-        // TODO: Can we poll the futures below at the same time?
-        // Access List + Name resolution and then Gas price + Gas
```

Comment:
I don't think we can poll the access list and name resolution at the same time, because the gas estimate and access list may be dependent on the `to` address.

It should be possible to poll gas price at the same time as any or all of the other futures, but would complicate the code a bit.